src/commands/contracts/call.ts

```diff
@@ -0,0 +1,83 @@
+import { createClient, createAccount } from "genlayer-js";
+import { simulator } from "genlayer-js/chains";
+import type { GenLayerClient } from "genlayer-js/types";
+import { getPrivateKey } from "../../lib/accounts/getPrivateKey";
+
+export interface CallOptions {
+  args: any[];
+  type: "read" | "write";
+}
+
+export class CallAction {
+  private genlayerClient: GenLayerClient<typeof simulator>;
+
+  constructor() {
+    this.genlayerClient = createClient({
+      chain: simulator,
+      endpoint: process.env.VITE_JSON_RPC_SERVER_URL,
+      account: createAccount(getPrivateKey() as any),
+    });
+  }
+
+  async call({
+               contractAddress,
+               method,
+               args,
+               type,
+             }: {
+    contractAddress: string;
+    method: string;
+    args: any[];
+    type: "read" | "write";
+  }): Promise<void> {
+    console.log(`Calling ${type} method ${method} on contract at ${contractAddress}...`);
+
+    try {
+      if (type === "read") {
```

Comment:
@epsjunior this could probably be inferred from the Intelligent contract schema, which you can get with getContractSchemaForCode: https://github.com/yeagerai/genlayer-js/blob/main/src/contracts/actions.ts#L13